app/decorators/valkyrie/resource_decorator.rb

```diff
@@ -187,6 +187,12 @@ def public_readable_state?
     workflow_class.public_read_states.include? Array.wrap(state).first.underscore
   end
 
+  # Is the resource embargoed?
+  # @return [TrueClass, FalseClass]
+  def embargoed?
+    EmbargoService.resource_embargoed?(model)
```

Comment:
Question: This is interesting - what led you to extract a class here instead of putting the function here and using the decorator in the ability?

Reply:
Or even putting it in Resource, although I'm not positive that has enough coverage.

Reply:
The decorator is better place for it. I was initially thinking I would add more methods to dry up date processing in a helper, but that doesn't seem necessary. Moved the method over.